### DISCUSSION TITLE
off-load git commands to smaller/cheaper models

### DISCUSSION THREAD

**Discussion by mfleurival** [cline/cline#884] [Feature Requests]
I'd like to request an update with a feature to set a custom Git command model. For example, I'd like to be able to use Sonnet 3.5 as my code editor, but run Haiku 3.5 with a different Git command. This would significantly reduce costs and improve overall efficiency.

Are there any concerns about memory loss or performance issues when switching between models during the same task? I've tried this before, but I'm not sure if it had a noticeable impact on the final result.

**Comment by BradKML** [comment]
Having multiple models in general is a good idea, the same idea for people bouncing between DeepSeek v3 and R1
